File: daily/922.按奇偶排序数组-ii.go
/*
 * @lc app=leetcode.cn id=922 lang=golang
 *
 * [922] 按奇偶排序数组 II
 */

// @lc code=start
func sortArrayByParityII(A []int) []int {
	j := 1;
	for i := 0; i < len(A) - 1; i+=2 {
		if A[i] & 1 == 1 {
			for A[j] & 1 == 1 { j += 2 }
			reflect.Swapper(A)(i, j)
		}
	}
	return A
}

// @lc code=end

File: daily/922.按奇偶排序数组-ii.rs
/*
 * @lc app=leetcode.cn id=922 lang=rust
 *
 * [922] 按奇偶排序数组 II
 */

// @lc code=start
impl Solution {
    pub fn sort_array_by_parity_ii(b: Vec<i32>) -> Vec<i32> {
        let mut a = b;
        let mut j = 1;
        for i in (0..=a.len() - 1).step_by(2) {
            if a[i] & 1 == 1 {
                while a[j] & 1 == 1 { j += 2; }
                a.swap(i, j);
            }
        }
        return a;
    }
}
// @lc code=end
